core/server/data/migrations/versions/3.18/01-add-email-preview-permissions-to-roles.js

```diff
@@ -0,0 +1,39 @@
+const utils = require('../../../schema/fixtures/utils');
+const {logging} = require('../../../../lib/common');
+
+const resource = 'email_preview';
+const _private = {};
+
+_private.getRelations = function getRelations(res) {
+    return utils.findPermissionRelationsForObject(res);
+};
+
+module.exports = {
+    config: {
+        transaction: true
+    },
+
+    async up(options) {
+        const relationFixtures = _private.getRelations(resource);
+        try {
+            const result = await utils.addFixturesForRelation(relationFixtures, options);
+            const success = result.done === result.expected;
+            if (!success) {
+                return logging.warn('Adding email_preview permissions to roles (did not insert)');
+            }
+            return logging.info('Adding email_preview permissions to roles');
+        } catch (err) {
+            return logging.error('Issue adding email_preview permissions to roles');
```

Comment:
We're still swallowing an error here - should we `throw` it, or avoid `catch`ing it so that the migrator can stop and display the error to the user, rather than continuing from an undefined state?

Reply:
@allouis Hmm don't think we are throwing errors anywhere in the migration files, there are quite a few places like [this](https://github.com/TryGhost/Ghost/blob/master/core/server/data/migrations/versions/3.0/11-update-posts-html.js#L34-L36) where we are swallowing it and resolving promise or just logging an error/warning. 

I am really not 100% what's the correct pattern here and possible downsides of starting throwing error in migration now, think thats something for later to sort though and keep this for now? 🤔

Reply:
That seems like specific logic to skip over things, in the same file you can see that we return a promise: https://github.com/TryGhost/Ghost/blob/master/core/server/data/migrations/versions/3.0/11-update-posts-html.js#L64-L67

This isn't "caught" so if the update errors, then the migration will stop!

Reply:
Hmm yeah good point, agree seems better to throw error here than let it go forward in that case, I'll update. 

Its definitely one of the bits to discuss though when we sort migrations in longer term, as we already have a mixed bag of handling for these cases 😬

Reply:
Updated now to throw error :)